### PR TITLE
Update Landing Page UI 

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*

*Description of changes:*
Updated the UI of the landing page to reflect finalized UI. Also added functionalities from the user stories, as well as snapshot test suite. 

TODOs in future: update snapshot tests away from mock data, and connect to back-end in future pull requests. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/components/main/report_definitions_table.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    field: 'owner',
+    name: 'Owner',
+    sortable: true,
+    truncateText: false,
+  },
+  {
+    field: 'source',
+    name: 'Source',
+    render: (username: string) => (
+      <EuiLink href={`https://github.com/${username}`} target="_blank">
```
Comment (seraphjiang) [inline public/components/main/report_definitions_table.tsx:48]: why use github.com user link here?

File: public/components/main/report_definitions_table.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  {
+    reportName: 'Daily Sales Report',
+    type: 'Schedule',
+    owner: 'davidcui',
+    source: 'dashboard/daily_saves',
+    lastUpdated: new Date('April 20 2020 19:32:102'),
+    details: 'Daily at 17:00:00.000 starting April 1, 2020',
+    status: 'Active',
+  },
+];
```
Comment (seraphjiang) [inline public/components/main/report_definitions_table.tsx:82]: move test data to separate file

File: public/components/main/report_definitions_table.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    }
+    actions={
+      <div>
+        <EuiButton
+          onClick={() => {
+            window.location.assign('opendistro-kibana-reports#/create');
+          }}
+        >
+          Create
+        </EuiButton>
```
Comment (seraphjiang) [inline public/components/main/report_definitions_table.tsx:105]: not required to change, but could we use EuiLink?
<EuiLink href={'opendistro-kibana-reports#/create'}>Create</EuiLink>

File: public/components/main/report_definitions_table.tsx (hunk 1 above)
Comment (davidcui-amzn) [reply public/components/main/report_definitions_table.tsx:48]: It was the default link from the Eui docs- can replace

File: public/components/main/report_definitions_table.tsx (hunk 2 above)
Comment (davidcui-amzn) [reply public/components/main/report_definitions_table.tsx:82]: Will do

File: public/components/main/report_definitions_table.tsx (hunk 3 above)
Comment (davidcui-amzn) [reply public/components/main/report_definitions_table.tsx:105]: In UI/UX flows we have this as a button instead of a link

**Pull request merged**